webapp/src/assetEditor.tsx

```diff
@@ -178,11 +179,38 @@ export class AssetEditor extends React.Component<{}, AssetEditorState> {
             kind: "ready"
         });
         tickAssetEditorEvent("asset-editor-shown");
+        this.pollingInterval = setInterval(this.pollForUpdates, 200);
```

Comment:
if we implemented an "onchange" function in the music editor and image editor, it might pick up changes faster and prevent the need for an unload event handler. i don't *think* it would need to be throttled at all...

Reply:
iirc we already listen for changes to the store in the image editor and the music editor has an on song change callback

Reply:
Yeah, that's probably the better way to do it, at first that's what I was thinking `tilemapProject.addChangeListener` would do until I looked a bit deeper; I think it's probably fine for now as even when I was attempting to induce it to save the wrong thing I was struggling to actual close the window in time after making a change such that this setInterval would miss (though if you have a pointer to those two events would be good to note down)